### PR TITLE
fix(delegation): bring back determinism

### PR DESCRIPTION
**Pull request by MaxMustermann2**
Unfortunately, serializing / marshalling a protobuf map into its bytes format is not guaranteed to be ordered.

### CODE REVIEW

**Review by bwhour — APPROVED** [review]

**Comment by cloud8little** [comment]
Test passed, three validators works well for exported testnet V1 data and start consensus.

**Review by mikebraver — APPROVED** [review]

**Review by TimmyExogenous — APPROVED** [review]
great work.

**Review by cloud8little — APPROVED** [review]

**Review by leonz789 — APPROVED** [review]

File: x/dogfood/keeper/impl_sdk.go [hunk 1]
```diff
@@ -214,7 +215,12 @@ func (k Keeper) IterateDelegations(
 	panic("unimplemented on this keeper")
 }
 
-func (k Keeper) WriteValidators(ctx sdk.Context) ([]tmtypes.GenesisValidator, error) {
+// WriteValidators returns all the currently active validators. This is called by the export
+// CLI. During this call, ctx.ChainID() is blank, and hence, chainID is passed separately.
+func (k Keeper) WriteValidators(
+	ctx sdk.Context, chainID string,
+) ([]tmtypes.GenesisValidator, error) {
+	fmt.Println("chainID", chainID)
```
Comment (cloud8little) [inline x/dogfood/keeper/impl_sdk.go]: remove the debug line plz.

Comment (MaxMustermann2) [reply x/dogfood/keeper/impl_sdk.go]: Reverted the commit. Will fix in a separate PR.

**Pull request merged**